data_structures/k_tree/find_matches.py

```diff
@@ -0,0 +1,18 @@
+"""Find matching values in k-nary tree."""
+
+
+def find_matches(tree, val):
+    """Find the matches."""
+    new = []
+
+    def find(node):
+        nonlocal new, val
```

Comment:
Since you never reassign these variables in this function, you do not need to declare them as `nonlocal`.